File: docs/vsto/codesnippet/VisualBasic/Trin_VstcoreActionsPaneWordVB/AddTextControl.vb
Public Class AddTextControl

    '--------------------------------------------------------------------------
    '<Snippet25>
    Private Sub insertText_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) _
        Handles insertText.Click

        With Globals.ThisDocument
            .InsertMemoFromNode.Text = Me.fromBox.Text
            .InsertMemoToNode.Text = Me.toBox.Text
            .InsertMemoSubjectNode.Text = Me.subjectBox.Text
        End With

        ' Clear the text boxes.
        Me.fromBox.Text = ""
        Me.toBox.Text = ""
        Me.subjectBox.Text = ""
    End Sub
    '</Snippet25>

End Class

File: docs/vsto/codesnippet/CSharp/Trin_VstcoreActionsPaneWordCS/AddTextControl.cs
using System;
using System.Windows.Forms;

namespace Trin_VstcoreActionsPaneWordCS
{
    partial class AddTextControl : UserControl
    {
        //---------------------------------------------------------------------
        //<Snippet26>
        public AddTextControl()
        {
            InitializeComponent();
            this.insertText.Click += new System.EventHandler(this.insertText_Click);
        }
        //</Snippet26>


        //---------------------------------------------------------------------
        //<Snippet25>
        private void insertText_Click(object sender, System.EventArgs e)
        {
            Globals.ThisDocument.InsertMemoFromNode.Text = this.fromBox.Text;
            Globals.ThisDocument.InsertMemoToNode.Text = this.toBox.Text;
            Globals.ThisDocument.InsertMemoSubjectNode.Text = this.subjectBox.Text;

            // Clear the text boxes.
            this.fromBox.Text = "";
            this.toBox.Text = "";
            this.subjectBox.Text = "";
        }
        //</Snippet25>
    }
}
